parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/ParallelConsumer.java

```diff
@@ -54,6 +55,8 @@
      */
     void poll(Consumer<ConsumerRecord<K, V>> usersVoidConsumptionFunction);
```

Comment:
FWIW my opinion of what the API could like is pretty much formulated here (and i like how it turned out):

https://github.com/mhowlett/howlett-kafka-extensions/blob/master/src/Howlett.Kafka.Extensions/Streaming/AsyncProcessor.cs

is it expected that the user will ever change the subscription after the processor is started? if not, should the subscription be specified on construction?

I don't like that the thing passed to the poll method is called a consumer, and the poll method is called poll. "registering" a "handler" would be better terminology.

Reply:
This is due to the Javiers construct for a consuming function being called a `Consumer` - I agree the naming isn't great. We could wrap the java `Consumer`, in order to rename it, but that would limit it's flexibility as all functions would have to inherit from out renamed Consumer.

Reply:
> is it expected that the user will ever change the subscription after the processor is started? if not, should the subscription be specified on construction?

Yeah this is a good point, and I've wrestled with it. One advantage of not having it in the constructor, is it leaves it open for `registering` other sorts of handlers after construction. It's certainly fathomable that this could happen. (of course, see your other comment).

Reply:
> FWIW my opinion of what the API could like is pretty much formulated here (and i like how it turned out):

Unfortunately my C-foo isn't up to scratch to really grok what the api is there :/